core/src/test/scala/kafka/server/metadata/MetadataPartitionsTest.scala

```diff
@@ -185,6 +184,134 @@ class MetadataPartitionsTest {
     assertEquals(2, updatedLeader.leaderId)
   }
 
+  @Test
+  def testTopicCreateAndDelete(): Unit = {
+    val topic = "foo"
+    val numPartitions = 3
+    val topicPartitions = (0 until numPartitions).map(new TopicPartition(topic, _)).toSet
+    val builder = new MetadataPartitionsBuilder(0, emptyPartitions)
+    val topicId = createTopic(topic, numPartitions, builder)
+    val localTopicPartitions = localChanged(builder)
+
+    assertTrue(localTopicPartitions.forall(topicPartitions.contains))
```

Comment:
How about `assertTrue(localTopicPartitions.subsetOf(topicPartitions))` (same to other test cases)?